lang/cs/Org.Apache.REEF.Network/Elastic/Operators/Physical/Default/DefaultOneToN.cs

```diff
@@ -0,0 +1,196 @@
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Threading;
+using System.Collections.Generic;
+using Org.Apache.REEF.Network.Elastic.Topology.Physical.Default;
+using System;
+using Org.Apache.REEF.Network.Elastic.Comm.Impl;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Utilities.Attributes;
+using Org.Apache.REEF.Network.Elastic.Operators.Physical.Enum;
+using Org.Apache.REEF.Network.Elastic.Comm;
+
+namespace Org.Apache.REEF.Network.Elastic.Operators.Physical.Default
+{
+    /// <summary>
+    /// Generic implementation of a group communication operator where one node sends to N.
+    /// </summary>
+    /// <typeparam name="T">The type of message being sent.</typeparam>
+    [Unstable("0.16", "API may change")]
+    public abstract class DefaultOneToN<T>
+    {
+        private static readonly Logger Log = Logger.GetLogger(typeof(DefaultOneToN<>));
+
+        internal readonly OneToNTopology _topology;
+        internal volatile PositionTracker _position;
+
+        private readonly bool _isLast;
+        private bool _cleanDisposal;
+
+        /// <summary>
+        /// Creates a new one to N operator.
+        /// </summary>
+        /// <param name="id">The operator identifier</param>
+        /// <param name="level">The checkpoint level for the operator</param>
+        /// <param name="isLast">Whether this operator is the last in the pipeline</param>
+        /// <param name="topology">The operator topology layer</param>
+        internal DefaultOneToN(int id, bool isLast, OneToNTopology topology)
+        {
+            OperatorId = id;
+            _isLast = isLast;
+            _topology = topology;
+            _position = PositionTracker.Nil;
+            _cleanDisposal = false;
+
+            OnTaskRescheduled = new Action(() =>
+            {
+                _topology.JoinTopology();
+            });
+        }
+
+        /// <summary>
+        /// The operator identifier.
+        /// </summary>
+        public int OperatorId { get; private set; }
+
+        /// <summary>
+        /// The operator type.
+        /// </summary>
+        public OperatorType OperatorType { get; protected set; }
+
+        /// <summary>
+        /// Operator-specific information that is sent to the driver in case of failure.
+        /// </summary>
+        public string FailureInfo
+        {
+            get
+            {
+                string iteration = IteratorReference == null ? "-1" : IteratorReference.Current.ToString();
+                string position = ((int)_position).ToString() + ":";
+                string isSending = _topology.IsSending ? "1" : "0";
+                return iteration + ":" + position + ":" + isSending;
```

Comment:
We should be able to do all that with proper format specifiers -- something along the lines of
```C#
return $"{IteratorReference?.Current ?? -1:d}:{_position:d}:{_topology.IsSending:d}";
```

Reply:
Much better and it fits in 100 chars. Fixed.